### PR TITLE
Document simplified filters context.

### PR DESCRIPTION
**Pull request by vbresults**
Ping @davidpersson 

https://github.com/UnionOfRAD/framework/pull/41#issuecomment-233122913

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -48,12 +48,16 @@
...
+  references to the static context are now made with `Context::class`, removing the need
+  for a static `$self` and repeating the full class name across filters in the same file.
+
   **Simplified filters signature.** By using PHP 5.4's new context binding feature for
   closures, we were able to simplify the signature of filters - i.e. by dropping the
   `$self`.
 
   This - as a sideeffect - reduces the requirement of using `invokeMethod()` to access
-  protected members of the context. `$this` and `static` can be used to access
+  protected members of the context. `$this` and `Context::class` can be used to access
```
Comment (davidpersson) [inline CHANGELOG.md]: Lets keep mentioning `static` as both approaches work. Something along the lines of "... $this and static/Context::class can be used..."

**Comment by davidpersson** [comment]
The following code example block will also need a change:
```
// always use this
Filters::apply('lithium\storage\Session'
```

**Comment by davidpersson** [comment]
Great that you found out about using "::class" this is way shorter and sweeter.

File: CHANGELOG.md (hunk 1 above)
Comment (vbresults) [inline CHANGELOG.md]: `static` does not work out of the box, it isn't automagically bound to the context class, and defaults to `lithium\core\Libraries` which in 99.9% of cases doesn't work the same;

https://github.com/UnionOfRAD/lithium/issues/1272

Comment (davidpersson) [inline CHANGELOG.md]: Yes, static can only be used when the filter was created inside a static method, they cannot later be bound so they can use `static``:
```
class Bar {

    protected static $_foo = 'foo';

    public static function giveFilter() {
        return function() {
            var_dump(static::$_foo);
        };
    }
}
```

`$this` will only work if the filter was created inside a concrete method OR you later bound the closure to an object.

My point here is that filters should mostly not have the need to interact directly with methods inside the class. They should be self contained. Yes there are some exceptions for these, you're right, let's not recommend `static` but `Context::class`.

Thanks for bringing this up.

**Comment by davidpersson** [comment]
Will merge this, okay?

**Pull request merged**